### PR TITLE
fix(angular): type resolution with strict templates

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [ ] Tests (`npm test`) were run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The generated Angular proxies do not take advantage of the new custom event interfaces as of Stencil 2.16.0. 

Issue URL: N/A

## What is the new behavior?

- Requires a peer dependency of `@stencil/core@2.16.0`
- Generates Angular output events with the new custom event interfaces
- Angular strict templates should no longer complain for output bindings

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

Requires Stencil 2.16.0. Developers will need to update their library dependency version to minimally `2.16.0`. 

## Other information

Dev-build: `0.6.1-dev.11655499531.137e1d26`

### CODE REVIEW

**Comment by sean-perkins** [comment]
This PR is a prerequisite to https://github.com/ionic-team/ionic-framework/pull/25502.

File: packages/angular-output-target/src/output-angular.ts [hunk 1]
```diff
@@ -103,7 +117,7 @@ import { ProxyCmp, proxyOutputs } from './angular-component-lib/utils';\n`;
 
       return `import { defineCustomElement as define${pascalImport} } from '${normalizePath(outputTarget.componentCorePackage!)}/${outputTarget.customElementsDir ||
         'components'
-      }/${component.tagName}.js';`;
+        }/${component.tagName}.js';`;
```
Comment (rwaskiewicz) [inline packages/angular-output-target/src/output-angular.ts:120]: Double checking - was this new white space intentional?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

File: packages/angular-output-target/src/generate-angular-component.ts [hunk 2]
```diff
@@ -56,6 +56,13 @@ export const createComponentDefinition = (
     '' // Empty first line
   ];
 
+  /**
+   * The list of @Output() bindings to generate for the component.
+   */
+  const outputBindings: string[]  = [
+    ''
```
[the remark this answers is not in the document]
Comment (sean-perkins) [reply packages/angular-output-target/src/generate-angular-component.ts]: Was used to create an empty new line when the output bindings are written to file, but I can see where that is very unclear with including an empty string in an array. 

I've changed this approach to remove the empty string and explicitly write a new line when pushing the output bindings to our `lines` array (which is later written to file).

**Review by rwaskiewicz — APPROVED** [review]

**Comment by sean-perkins** [comment]
Closing this PR in favor of https://github.com/ionic-team/stencil-ds-output-targets/pull/308. Will open that PR for review after generating a dev-build and doing one more pass of testing.

**Pull request closed without merging**